src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/Dependencies/AssemblyReference.xaml

```diff
@@ -93,4 +93,7 @@
                 ReadOnly="True"
                 Visible="False" />
 
+  <BoolProperty Name="TreatAsUsed"
+                Description="Indicates that this reference should be treated as if it were used."/>
```

Comment:
Is this supposed to be visible to the user in the properties pane? If so, it must have `DisplayName`. If not, set `Visible` to false and remove the `Description`. Ditto other rules in this PR.

Reply:
These have to be added to the ResolvedXXX versions of these rules. I'm surprised we don't have tests for this.

Reply:
I think we should sync with .NET Core reference trimming feature to see if they have a similar feature for this (to indicate that a reference should not be removed during trimming), so there's only one concept for this.

Reply:
Done. Added to ResolvedProjectReference, ResolvedPackageReference, ResolvedAssemblyReference and removed from SdkReference.